### ISSUE TITLE
Add auto save on event, i.e: VimExit?

### ISSUE THREAD

**Issue by tmpm697** [gennaro-tedesco/nvim-possession#6]
manually saving can be tedious, how about auto-save when VimExit?

**Comment by gennaro-tedesco** [comment]
Yes, this can be done too, I will look into it. 

> can just use some kind of default from nvim like

I would ideally implement this for the cases when a session is loaded already, so that a name and a session file already exists and it's evident the user is still working on it. As such, what is needed is just an autocommand to re-save the session on VimExit.

**Comment by tmpm697** [comment]
yes, autocmd to watch on `PreDirChange` and `DirChange` would cover switching sessions and `VimExit` should cover exit vim event.

**Comment by gennaro-tedesco** [comment]
Correct, that shouldn't be too hard, I will give it a go later!

**Comment by gennaro-tedesco** [comment]
Autosave mechanism is now merged on `main` after https://github.com/gennaro-tedesco/nvim-possession/releases/tag/v0.0.3.

Still to work on the switch and dir change, but we are getting there :)
